### PR TITLE
Polymer2.0 preview

### PR DESCRIPTION
**Pull request by davidrleonard**
@mdwragg @randyaskin This is ready to merge with all Polymer 2.x file, test, and demo updates. This is going to be a major to deprecate and remove a lot of old px-card view service behaviors. Plz review.

Changes as described in HISTORY.md below:

v2.0.0
==================
Upgrades <px-card> element and its related styles to support Polymer 1 and 2.
Refactors and removed deprecated code. There are many major breaking changes
in this release, please see below for more information and an upgrade guide:

## Breaking change: px-card view service behaviors and components deprecated

px-card is now a presentation component only. All code related to managing
decks, dashboards, dealers, and communications with the Predix View Service
have been removed as of this version.

The following components and behaviors have been deprecated and their files deleted:

* `window.px.card` behavior removed
* `window.px.dealer` behavior removed
* `window.px.dashboard` behavior removed
* `<px-dashboard>` component removed
* px-card/px-dashboard.html file deleted
* `window.px.deck` behavior removed
* `<px-deck>` component removed
* px-card/px-deck.html file deleted
* `<px-sample-card>` component removed
* px-card/px-sample-card.html file deleted

Search your code to ensure you do not import any deleted files or rely on any
removed behaviors/components.

**Upgrade path:** To continue using these behaviors, do not upgrade to the newest
px-card version and remain on the last major release.

## Breaking change: px-card actions slot API
The px-card slot API has changed to support the webcomponents v1 specification
for passing children into a custom element.

The previous version of px-card allowed developers to pass in a div with the
class "actions" to place icon buttons into the px-card header:

```
<px-card>
  <div class="actions">
    <px-icon icon="..."></px-icon>
    <px-icon icon="..."></px-icon>
    <px-icon icon="..."></px-icon>
  </div>
  <p>Card text goes here.</p>
</px-card>
```

**Upgrade path:** To upgrade, developers should remove the div with the "actions"
class and instead put a new `slot="actions"` attribute directly on each <px-icon>
tag. This is the same code as above with the updates:

```
<px-card>
  <px-icon slot="actions" icon="..."></px-icon>
  <px-icon slot="actions" icon="..."></px-icon>
  <px-icon slot="actions" icon="..."></px-icon>
  <p>Card text goes here.</p>
</px-card>
```

## Breaking change: px-card header always shown unless configured

Previously, not setting the `headerText` property caused the px-card to hide
its header, including the icon and actions areas. The documentation recommended
passing in a single space to the `headerText` property to show the icon and
actions areas without showing any header text.

This behavior has been changed. Not setting the `headerText` property no longer
hides the px-card header. To hide the header, set the `hideHeader` property to
true. It defaults to false.

**Upgrade path:** Ensure your code did not rely on this px-card behavior and does
not expect cards with no header text to hide their headers. If your code does
rely on this behavior, you should decide whether or not to hide the header for
any px-card elements and set the `hideHeader` property to true to hide the header.

You should also remove any instances of setting the px-card header text to a
string with a single space (e.g. `header-text=" "`). This is no longer needed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ temp
 bower_components
 .idea
 reports
-css/noprefix/
+css/noprefix
```
Comment (mdwragg) [inline .gitignore]: We can drop this - proper old, back when we were using grunt!

File: examples/card.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>px-card demo</title>
+
+  <!-- Load webcomponentsjs + Polymer -->
+  <script src="../webcomponentsjs/webcomponents-loader.js"></script>
```
Comment (mdwragg) [inline examples/card.html]: Switch out to conditional webcomponents-lite.js loading to keep Polymer 1.x working.

File: examples/card.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <link rel="import" href="../polymer/polymer.html"/>
+
+  <!-- Load theme -->
+  <link rel="import" href="../px-theme/px-theme-styles.html"/>
+  <custom-style>
+    <style include="px-theme-styles"></style>
```
Comment (mdwragg) [inline examples/card.html]: Needs `is="custom-style"` here to keep working on Polymer 1.x

**Comment by mdwragg** [comment]
Obvs, make sure CI is happy prior to merge ;)

Few comments in the review but looks sweet. Feels great to trim that much code!

💯

**Comment by davidrleonard** [comment]
All those fixes are in, Travis is failing randomly because the Sauce tunnel won't open. We'll see if this round passes then i'll merge/tag/release.

**Pull request merged**